app/models/user.rb

```diff
@@ -1,2 +1,9 @@
 class User < ActiveRecord::Base
+  before_save { self.email = email.downcase }
```

Comment:
[IMO] The email always do be downcased even it is not changed. Please try to using before_create or something to avoid that :bowing_man: